posthog/models/property.py

```diff
@@ -15,10 +15,21 @@
 from posthog.constants import PropertyOperatorType
 from posthog.models.filters.mixins.utils import cached_property
 from posthog.models.filters.utils import GroupTypeIndex, validate_group_type_index
-from posthog.utils import is_valid_regex
+from posthog.utils import is_valid_regex, str_to_bool
 
 ValueT = Union[str, int, List[str]]
-PropertyType = Literal["event", "person", "cohort", "element", "static-cohort", "precalculated-cohort", "group"]
```

Comment:
Did not change this to an enum. It requires a pretty extensive retyping. There's a case where we're using breakdown_type as the value to initialize a Property object with which breaks the enum pattern because we'd have to retype breakdown_type too